AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -178,17 +178,14 @@ - (AWSTask *)saveRecord:(NSData *)data
                                ];
                 if (!result) {
                     AWSLogError(@"SQLite error. [%@]", db.lastError);
-                    error = db.lastError;
+                    dbError = db.lastError;
                 }
             }];
         }
-
-        if (error) {
```

Comment:
That indeed makes sense, @yosuke-matsuda 
PR has been updated to reflect this strategy